### PR TITLE
Refactor RecipeRepository class and test

### PR DESCRIPTION
**Pull request by aliroemhildt**
#### What's this PR do?
refactor RecipeRepository class and test
refactor Recipe class and test

#### Where should the reviewer start?
the top of each file, several changes throughout

#### How should this be manually tested?
run npm test, all tests are still passing

#### Any background context you want to provide?
removed ingredientsData as a class property for each of these classes, made it a param for the methods instead
ingredientsData will be a global var in our main.js

#### What are the relevant tickets?
n/a

#### Screenshots (if appropriate)
n/a

#### Questions:
n/a

### CODE REVIEW

File: src/classes/RecipeRepository.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class RecipeRepository {
-  constructor(recipeData, ingredientsData) {
+  constructor(recipeData) {
     this.recipeData = recipeData;
-    this.ingredientsData = ingredientsData;
+    this.recipesToShow = [];
```
Comment (anewb87) [inline src/classes/RecipeRepository.js:4]: Like this idea, I think this will serve us well down the road.

**Review by kammon10 — APPROVED** [review]
Crushed it!!

File: test/RecipeRepository-test.js [hunk 2]
```diff
@@ -341,7 +341,7 @@ describe('RecipeRepository', () => {
         ]
       }
     ]
-    repository = new RecipeRepository(sampleData, ingredientsData);
+    repository = new RecipeRepository(sampleData);
```
Comment (anewb87) [inline test/RecipeRepository-test.js:344]: Re-organized our test and method.

**Review by anewb87 — APPROVED** [review]

**Review by cagallo — APPROVED** [review]
This is great, makes more sense to pass this data through in the methods! I really like the new array with recipes to show bc that will be dynamic depending on what the user is viewing.

**Pull request merged**